From 487fb384e2fe9c3c5752c31a1cdb24cd1718b72c Mon Sep 17 00:00:00 2001
From: letsfindaway <me@letsfindaway.de>
Date: Mon, 18 Nov 2024 08:20:33 +0100
Subject: [PATCH] fix: avoid some wayland problems

- use QT_QPA_PLATFORM=xcb when running on wayland
- avoids crash when using the screen configuration list
- avoids invisible window on Qt 6.8
---
 src/core/main.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/core/main.cpp b/src/core/main.cpp
index f9dbce5b1..2a978ff8e 100644
--- a/src/core/main.cpp
+++ b/src/core/main.cpp
@@ -96,6 +96,14 @@ int main(int argc, char *argv[])
     if (qEnvironmentVariableIsSet("QT_NO_GLIB"))
         qunsetenv("QT_NO_GLIB");
 
+    // force using xwayland when running on wayland
+#ifdef Q_OS_LINUX
+    if (qgetenv("XDG_SESSION_TYPE") == "wayland")
+    {
+        qputenv("QT_QPA_PLATFORM", "xcb");
+    }
+#endif
+
     Q_INIT_RESOURCE(OpenBoard);
 
     qInstallMessageHandler(ub_message_output);
